endgame_simulations/simulations.py

```diff
@@ -233,15 +233,11 @@ def iter_run(
             total=real_end_time - self.state.current_time + self._delta_time,
             disable=not self.verbose,
         ) as progress_bar:
-            while self.state.current_time <= real_end_time:
-                if self.state._previous_delta_time is None:
-                    prev_delta_time = self._delta_time
-                else:
-                    prev_delta_time = self.state._previous_delta_time
-
+            while self.state.current_time + self._delta_time <= real_end_time:
```

Comment:
So the current logic in the code is, after a change of parameter set (and therefore possibly a change of timestep), to perform the first iteration with the previous timestep. Do you understand why? Do you think its purely a mistake?

Reply:
I think it is purely just because a mistake - missing an edge case that would occur when changing the delta_time (timestep) parameter. In the cases where we don't change this parameter, we wouldn't see any bug, but because we change the timestep parameter mid simulation, it causes an issue.